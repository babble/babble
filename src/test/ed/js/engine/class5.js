
/**
*    Copyright (C) 2008 10gen Inc.
*  
*    This program is free software: you can redistribute it and/or  modify
*    it under the terms of the GNU Affero General Public License, version 3,
*    as published by the Free Software Foundation.
*  
*    This program is distributed in the hope that it will be useful,
*    but WITHOUT ANY WARRANTY; without even the implied warranty of
*    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*    GNU Affero General Public License for more details.
*  
*    You should have received a copy of the GNU Affero General Public License
*    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

function Person( name ){
    this._name = name;
}

Person.prototype.something = function( a ){
    print( a )
};

Person.prototype.printName = function(){
    print( this._name );
}

var p = new Person( "eliot" );


Person.prototype.a = "before";
print( Person.prototype.a );
print( p.a );


Person.prototype.a = "after";
print( p.a );

print( p._name );
p.something("yo");

p.printName();

p.foo = function() { 
    print( this._name );
};
p.foo();


a = {};
a.a = {};
a.a.b = function(){
    this.z = 1;
};

foo = new a.a.b();
print( foo.z );


function Post() {
    this.x = 5;
};

ns = {Post: Post};

p1 = new Post();
p2 = new ns.Post();
print(p1.x);
print(p2.x);

function makeArray() {
    return arguments;
}

print( (new makeArray( 1,2,3 )).length );
print( (new makeArray()).length );
