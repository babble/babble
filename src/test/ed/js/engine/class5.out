before
before
after
eliot
yo
eliot
eliot
1
5
5
